api/src/main/java/com/clubobsidian/dynamicgui/api/effect/ParticleWrapper.java

```diff
@@ -42,17 +42,32 @@ public ParticleWrapper(@NotNull ParticleData data) {
         this.data = Objects.requireNonNull(data);
     }
 
+    /**
+     * Gets the ParticleData for the particle effect
+     *
+     * @return the ParticleData object
+     */
     public ParticleData getData() {
         return this.data;
     }
 
+    /**
+     * Spawns the particle effect for the player
+     *
+     * @param player the player to spawn the effect for
+     */
     public void spawnEffect(@NotNull PlayerWrapper<?> player) {
         Objects.requireNonNull(player);
         player.playEffect(this.data);
     }
 
     public static class ParticleData {
 
+        /**
+         * Creates a ParticleData object from a string
+         * @param str the string to create the ParticleData object from
+         * @return
```

Comment:
Add return comment, something along the lines of "Returns ParticleData from the given string"